app/models/user.rb

```diff
@@ -9,4 +9,55 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  has_many :friendships
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
+
+  def friends
+    friends_array = friendships.map { |friendship| friendship.friend if friendship.confirmed }
+    friends_arrayb = inverse_friendships.map { |friendship| friendship.user if friendship.confirmed }
+    friends_array.concat(friends_arrayb)
+    friends_array.compact
+  end
```

Comment:
**Optional** You don't need this method if you implemented the associations correctly. You can easily check for the `friendships` and `inverse_friendships` of a user. It is not a good practice to use Enumerable to retrieve objects from the database. 

HINT: add the `where` option to only select those that match `confirmed: true`.
```
Scopes
You can pass a second argument scope as a callable 
(i.e. proc or lambda) to retrieve a specific set of records or 
customize the generated query when you access the associated collection.

Scope examples:

has_many :comments, -> { where(author_id: 1) }
has_many :employees, -> { joins(:address) }
has_many :posts, ->(blog) { where("max_post_length > ?", blog.max_post_length) }
```